src/server/pps/server/api_server.go

```diff
@@ -975,6 +975,9 @@ func (a *apiServer) GetLogs(request *pps.GetLogsRequest, apiGetLogsServer pps.AP
 	// (based on pipeline and job filters)
 	var rcName, containerName string
 	if request.Pipeline == nil && request.Job == nil {
+		if len(request.DataFilters) > 0 || request.Datum != nil {
+			return fmt.Errorf("must specify the Job or Pipeline that the datum is from to get logs for it")
+		}
```

Comment:
Makes sense. Do we have test coverage for these cases?

Reply:
Not for the error case I don't think. I can add that.